src/lib.rs

```diff
@@ -269,7 +276,7 @@ impl Store {
     // }
     const fn new()-> Self {
         // let mut self_ =
-        const W: usize = PAGE_SIZE/mem::size_of::<Pair>();
+        const W: usize = PageType::Pair.items_per_page();
         const ZEROED: Pair = Pair([0,0]);
```

Comment:
hm, this seems much less _direct_ for "`W` such that `mem::size_of::<[Pair; W]>()` is `PAGE_SIZE`"? For `rc` it makes more sense, though there I'd probably put a `const max_items` on `PageType` directly, instead of floating around "well `Pair` is the smallest one"

Reply:
I actually think it's more direct for `pages` than for `rc` . It makes sense to me that a newly-initialized `Pages` should have each page comprised of as many `ZEROED` pairs as there are pairs per page, which was just assigned to the variable `W` for brevity.  Whereas it's not as clear why the `rc` arrays should be of length `Pair.items_per_page` without knowing that `Pair` is the smallest one. `PageType::max_items` makes sense to me.

Reply:
…okay now this line doesn't mention `Pair` at /all/

Reply:
How do you feel about:

```
    const fn new()-> Self {
        const ZEROED: Pair = Pair([0,0]);
        Self {
            types: [None; PAGES],
            pages: Pages { pairs: [const{[ZEROED; PageType::Pair.items_per_page()]}; PAGES] }, // basically mem::zeroed()
            rc: [[0; PageType::max_items()]; PAGES],
            free: [true; PAGES],
            full: [false; PAGES],
        }
    }

```
?

Reply:
Hm does it work to
```rs
        const ZEROED = [Pair([0,0]); PageType::Pair.items_per_page()]
        Self {
            types: [None; PAGES],
            pages: Pages { pairs: [ZEROED; PAGES] }, // basically mem::zeroed()
            rc: [[0; PageType::max_items()]; PAGES],
            free: [true; PAGES],
            full: [false; PAGES],
        }
```
actually?

Reply:
Alas no:
```
error[E0308]: mismatched types
   --> src/lib.rs:288:36
    |
288 |             pages: Pages { pairs: [ZEROED; PAGES] }, // basically mem::zeroed()
    |                                    ^^^^^^ expected array of 16 elements, found struct `Pair`
    |
    = note: expected array `[Pair; 16]`
              found struct `Pair`

```

Reply:
I think I'd take `pages: Pages { pairs: [const{[ZEROED_PAIR; PageType::Pair.items_per_page()]};` but not like, consider it an improvement on the status quo :P

Reply:
Yeah I'm saying make ZEROED the whole page array

Reply:
This seems to do the trick:
```
    const fn new()-> Self {
        const W: usize = PageType::Pair.items_per_page();
        const ZEROED: [Pair; W]  = [const{Pair([0,0])}; W];
        Self {
            types: [None; PAGES],
            pages: Pages { pairs: [ZEROED; PAGES] }, // basically mem::zeroed()
            rc: [[0; PageType::max_items()]; PAGES],
            free: [true; PAGES],
            full: [false; PAGES],
        }
    }
```